src/test/java/com/gfl/client/model/ProxyConfigHolderTest.java

```diff
@@ -0,0 +1,195 @@
+    void testHashCode() {
+        int expected = proxyConfigHolder.hashCode();
+        assertEquals(expected, proxyConfigHolder.hashCode());
+    }
+
+    @Test
+    @DisplayName("Test toString method for proxyConfigHolder object")
+    void testToString() {
+        String expected = "ProxyConfigHolder(proxyNetworkConfig=" +
+                "ProxyNetworkConfig(hostname=hostname, port=8080)," +
+                " proxyCredentials=ProxyCredentials(username=username, password=password))";
+        assertEquals(expected, proxyConfigHolder.toString());
+    }
+
+    @Test
+    @DisplayName("Test equality and hashCode for   ProxyConfigHolder objects with the same values")
+    void testEqualsAndHashCode() {
+        ProxyConfigHolder proxyConfigHolder1 = new ProxyConfigHolder();
+        ProxyNetworkConfig config = new ProxyNetworkConfig();
+        config.setHostname("hostname");
+        config.setPort(8080);
+        ProxyCredentials credentials = new ProxyCredentials();
+        credentials.setUsername("username");
+        credentials.setPassword("password");
+
+        proxyConfigHolder1.setProxyCredentials(credentials);
+        proxyConfigHolder1.setProxyNetworkConfig(config);
+
+        assertEquals(proxyConfigHolder, proxyConfigHolder1);
+        assertEquals(proxyConfigHolder.hashCode(), proxyConfigHolder1.hashCode());
+
+    }
+
+    @Test
+    @DisplayName("Test validation fails for null proxyNetworkConfig")
+    void testValidationFailsForNullProxyNetworkConfig() {
+        ProxyConfigHolder configHolder = new ProxyConfigHolder(null, proxyCredentials);
+
+        Set<ConstraintViolation<ProxyConfigHolder>> violations = validator.validate(configHolder);
+
+        // Assert violations for null proxyNetworkConfig
+        assertEquals(1, violations.size());
+        ConstraintViolation<ProxyConfigHolder> violation = violations.iterator().next();
+        assertEquals("proxy network configuration can't be null", violation.getMessage());
+        assertEquals("proxyNetworkConfig", violation.getPropertyPath().toString());
+    }
+
+    @Test
+    @DisplayName("Test validation fails for invalid proxyCredentials")
+    void testValidationFailsForInvalidProxyCredentials() {
+        ProxyConfigHolder configHolder = new ProxyConfigHolder(proxyNetworkConfig, new ProxyCredentials(null, null)); // Assuming null values are invalid
+
+        Set<ConstraintViolation<ProxyConfigHolder>> violations = validator.validate(configHolder);
+
+        // Asserts the total number of violations -> Assert violations for invalid proxyCredentials
+        assertEquals(2, violations.size()); // Two violations for null username and password
+
+        // Verify violation for null username
+        ConstraintViolation<ProxyConfigHolder> usernameViolation = violations.stream()
+                .filter(v -> v.getPropertyPath().toString().equals("proxyCredentials.username")) // Verifies specific violations for username and password using property paths.
```

Comment:
Better use vice versa "proxyCredentials.password".equals(v.getPropertyPath().toString())
Is a good practice, especially when dealing with string literals. 
It helps avoid potential NullPointerException if getPropertyPath() returns null and makes the code more robust.